File: Back-end/controllers/mailController.js
const { Resend } = require("resend");
const dotenv = require("dotenv");

dotenv.config();

const resend = new Resend(process.env.RESEND_API_KEY);

const sendMail = async (req, res) => {
  const { nombre, email, mensaje } = req.body;

  if (!nombre || !email || !mensaje) {
    return res.status(400).json({ error: "Todos los campos son obligatorios" });
  }

  try {
    await resend.emails.send({
      from: "Tu Sitio <onboarding@resend.dev>", // puedes cambiarlo luego por tu dominio verificado
      to: process.env.EMAIL_USER, // tu correo personal o de recepción
      subject: `Nuevo mensaje de ${nombre}`,
      html: `
        <h3>Detalles del contacto</h3>
        <p><b>Nombre:</b> ${nombre}</p>
        <p><b>Email:</b> ${email}</p>
        <p><b>Mensaje:</b> ${mensaje}</p>
      `,
    });

    res.status(200).json({ message: "Correo enviado con éxito 🚀" });
  } catch (error) {
    console.error("Error al enviar correo:", error);
    res.status(500).json({ error: "Error al enviar correo" });
  }
};

module.exports = { sendMail };

File: Back-end/controllers/mail_controller.py
import os
import requests
from flask import Blueprint, request, jsonify

mail_bp = Blueprint("mail", __name__)


@mail_bp.route("/", methods=["POST"])
def send_mail():
    """Envía un correo usando la API de Resend con los datos de contacto."""
    payload = request.get_json(silent=True) or {}
    nombre = payload.get("nombre")
    email = payload.get("email")
    mensaje = payload.get("mensaje")

    if not nombre or not email or not mensaje:
        return jsonify({"error": "Todos los campos son obligatorios"}), 400

    api_key = os.getenv("RESEND_API_KEY")
    to_address = os.getenv("EMAIL_USER")
    if not api_key or not to_address:
        return jsonify({"error": "Configuración de correo incompleta"}), 500

    body = {
        "from": "Tu Sitio <onboarding@resend.dev>",
        "to": [to_address],
        "subject": f"Nuevo mensaje de {nombre}",
        "html": f"<h3>Detalles del contacto</h3><p><b>Nombre:</b> {nombre}</p><p><b>Email:</b> {email}</p><p><b>Mensaje:</b> {mensaje}</p>",
    }

    try:
        resp = requests.post(
            "https://api.resend.com/emails",
            headers={"Authorization": f"Bearer {api_key}", "Content-Type": "application/json"},
            json=body,
            timeout=15,
        )
        if resp.status_code >= 400:
            return jsonify({"error": "Error al enviar correo", "details": resp.text}), 502
        return jsonify({"message": "Correo enviado con éxito 🚀"}), 200
    except requests.RequestException as e:
        return jsonify({"error": "Fallo de red al enviar correo", "details": str(e)}), 500